policy/data/subscription.go

```diff
@@ -0,0 +1,40 @@
+package data
+
+import (
+	"context"
+	"fmt"
+	"olympos.io/encoding/edn"
+)
+
+// SubscriptionDataSource allows querying of objects directly from the subscription results array.
+// It is beneficial to use this data source when it is useful to query by name,
+// possibly allowing earlier data sources to pick up the query first.
+type SubscriptionDataSource struct {
+	queryIndexes        map[string]int
+	subscriptionResults [][]edn.RawMessage
+}
+
+func NewSubscriptionDataSource(queryIndexes map[string]int, subscriptionResults [][]edn.RawMessage) SubscriptionDataSource {
+	return SubscriptionDataSource{
+		queryIndexes:        queryIndexes,
+		subscriptionResults: subscriptionResults,
+	}
+}
+
+func (ds SubscriptionDataSource) Query(_ context.Context, queryName string, _ string, _ map[string]interface{}, output interface{}) (*QueryResponse, error) {
+	ix, ok := ds.queryIndexes[queryName]
```

Comment:
Maybe this is just personal preference, but I find it much harder to read when every entity is abbreviated.